packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -351,7 +405,14 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
       }
     }
     if (event is PointerUpEvent || event is PointerCancelEvent || event is PointerPanZoomEndEvent) {
-      _giveUpPointer(event.pointer);
+      if (_pointerMoveAccept) {
+        _velocityTrackers.keys.toList().forEach(_giveUpPointer);
```

Comment:
```suggestion
        // Clone the list with `toList` because `_giveUpPointer` modifies the list.
        _velocityTrackers.keys.toList().forEach(_giveUpPointer);
```